pilot/pkg/model/service.go

```diff
@@ -53,10 +53,23 @@ type Service struct {
 	// Address specifies the service IPv4 address of the load balancer
 	Address string `json:"address,omitempty"`
 
+	// ExternalAddresses are external IPv4 or hostname addresses for accessing
+	// the service. They are not the VIP and might not be set. E.g. in
+	// Kubernetes, if the service is of type LoadBalancer, its public
+	// address(es) will be placed in this field
+	ExternalAddresses []string `json:"external-addresses,omitempty"`
+
 	// ClusterVIPs specifies the service address of the load balancer
 	// in each of the clusters where the service resides
 	ClusterVIPs map[string]string `json:"cluster-vips,omitempty"`
 
+	// ClusterExternals will hold the mapping between the Cluster ID and the
+	// external address(es) for the service that resides within that cluster.
+	// When a service with the same hostname exists in multiple clusters, the
+	// discovery will return a single service entry that aggregates the
+	// external address(es) for each cluster in this field.
+	ClusterExternals map[string][]string `json:"cluster-externals,omitempty"`
+
```

Comment:
This is not the right way. It will break all future efforts. Instances attach to a service. Not the other way around. Move this to the network endpoint. Endpoints should have a network name.

Reply:
Agree Endpoints should have network name, and ok to put it now in network endpoint - but I'm trying to stop 'instances attach to service' and the List interface. Will worry about this when my incremental EDS is merged.

Reply:
I mean - the network name (nid ?) will be in the endpoint. 
The map of external addresses for each network should probably be a private field in XDS server, 
it doesn't need to be in model.
Service can have the list of 'external'/public IPs.

Reply:
Yes. But the general principle applies. Network names in endpoints. Not sticking them into the service!